tests/test_documents.py

```diff
@@ -0,0 +1,381 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# This file is formatted with Python Black
+
+import tests as xdp
+
+import pytest
+import dbus
+from pathlib import Path
+import os
+from gi.repository import GLib
+
+
+def path_from_null_term_bytes(bytes):
+    path_bytes = bytes.split(b"\x00")[0]
+    return Path(os.fsdecode(path_bytes))
+
+
+def get_mountpoint(documents_intf):
+    mountpoint = documents_intf.GetMountPoint(byte_arrays=True)
+    mountpoint = path_from_null_term_bytes(mountpoint)
+    assert mountpoint.exists()
+    return mountpoint
+
+
+def write_bytes_atomic(file_path, bytes):
+    GLib.file_set_contents(file_path.absolute().as_posix(), bytes)
+
+
+def write_bytes_trunc(file_path, bytes):
+    try:
+        fd = os.open(
+            file_path.absolute().as_posix(), os.O_RDWR | os.O_TRUNC | os.O_CREAT
+        )
+        os.write(fd, bytes)
+    finally:
+        os.close(fd)
+
+
+def export_file(documents_intf, file_path, unique=False):
+    assert file_path.exists()
+
+    with open(file_path.absolute().as_posix(), "r") as file:
+        doc_id = documents_intf.Add(file.fileno(), not unique, False)
+        assert doc_id
+
+    return doc_id
+
+
+def export_file_named(documents_intf, folder_path, name, unique=False):
+    assert folder_path.exists()
+
+    # why does this need to be NULL terminated? otherwise:
+    # dbus.exceptions.DBusException: org.freedesktop.portal.Error.InvalidArgument
+    name_nt = name.encode("utf8") + b"\x00"
```

Comment:
This is necessary because when you give dbus-python a `bytes` object, it assumes you want to send exactly those bytes. It does not know about the "bytestring" convention that originated in GVariant (representing "text in an unspecified encoding" as the bytes + `\0`), which is newer than the dbus-python API design.

This should probably be `os.fsencode(path) + b"\x00"` to handle being run from a directory whose name is in a legacy non-UTF-8 encoding.